### PR TITLE
Implement plan/apply/destroy for landingzones

### LINKED ISSUE #21: Implement login_as_launchpad
Specifically the extraction of TF_VAR_tfstate_storage_account_name etc vars from the keyvault

https://github.com/aztfmod/rover/blob/1113d610f00c2bc5b2d80d970e66052acd1d5ed3/scripts/functions.sh#L266

This is a broad tasks and might require breaking up, further investigation is required

### PR DESCRIPTION
**Pull request by benc-uk**
This is a small but very important PR it enables `landingzone` and `cd` operations for levels 1+. This means that plan/apply/destroy actions can now function on landingzone levels 1+ 

e.g. all of the following will now work....
* `cd apply --symphony-config local/symphony.yaml --level level1`
* `cd apply --symphony-config local/symphony.yaml`  (all levels)
* `cd destroy --symphony-config local/symphony.yaml --level level1`
* `landingzone apply --config-path ./local/configs/level1/web --source ./landingzones --statename web`

The key part making this happen is the function `connectToLaunchPad()` which sets many `TF_VAR_` variables based on the values extracted from the KeyVault located in the launchpad

The bulk of the important changes are in this commit
https://github.com/aztfmod/rovergo/pull/60/commits/265e30c347461b4eba81c0f55cab05e30528f3b5 

Fixes #60  
Fixes #21

### CODE REVIEW

File: pkg/landingzone/landingzone.go [hunk 1]
```diff
@@ -384,3 +394,62 @@ func (o *Options) enableAzureBackend() {
...
+
+	_, _, keyVaultName := azure.ParseResourceID(lpKeyVaultID)
+
+	// HACK: Need to support other Azure cloud endpoints
+	kvClient, err := azure.NewKVClient("vault.azure.net", keyVaultName)
+	if err != nil {
+		return err
+	}
+
+	lpTenantID, err := kvClient.GetSecret("tenant-id")
```
Comment (sebastus) [inline pkg/landingzone/landingzone.go:417]: Define constants for the keyvault keys?

File: pkg/landingzone/landingzone.go [hunk 2]
```diff
@@ -384,3 +394,62 @@ func (o *Options) enableAzureBackend() {
...
+	if err != nil {
+		return err
+	}
+	lpLowerResGrp, err := kvClient.GetSecret("lower-resource-group-name")
+	if err != nil {
+		return err
+	}
+
+	if lpLowerSAName == "" || lpTenantID == "" || lpLowerResGrp == "" {
+		return fmt.Errorf("Required secret(s) not found in launchpad, either you are not authorized ot the launchpad was not deployed correctly")
```
Comment (sebastus) [inline pkg/landingzone/landingzone.go:431]: spelling

File: pkg/landingzone/landingzone.go [hunk 3]
```diff
@@ -384,3 +394,62 @@ func (o *Options) enableAzureBackend() {
 	err := utils.CopyFile(o.SourcePath+"/backend.azurerm", o.SourcePath+"/backend.azurerm.tf")
```
Comment (sebastus) [inline pkg/landingzone/landingzone.go:394]: filepath.Join...?

File: readme.md [hunk 4]
```diff
@@ -4,83 +4,43 @@ A project to undertake re-writing the [Rover tool](https://github.com/aztfmod/ro
 
 Uses [Cobra](https://github.com/spf13/cobra) to provide the framework for a robust and familiar CLI tool and [Viper](https://github.com/spf13/viper) for configuration
 
+This is a breaking change from the previous version with large scale changes.
+
 # Current Status
 
 ### [👷‍♂️ Project board](https://github.com/orgs/aztfmod/projects/28?card_filter_query=label%3Arover-go)
 ### ☢ This is under heavy development, expect braking changes almost daily 🔥
```
Comment (sebastus) [inline readme.md:12]: I guess breaking changes really are braking as well. ;)

File: readme.md [hunk 5]
```diff
@@ -4,83 +4,43 @@ A project to undertake re-writing the [Rover tool](https://github.com/aztfmod/ro
...
-- Shape of commands & sub-commands and CLI structure
-- `launchpad fetch` - Implements the existing `--clone` option
-- `launchpad run` - To deploy a launchpad
-  - Actions init, plan, deploy & destroy implemented and working
+- `launchpad` - To deploy a launchpad
+  - Actions init, plan, apply, destroy, fmt & validate implemented and working
   - Handling of state initialization and upload
-  - Handling of locating remote state from level and CAF environment
-- `cd run` - To run actions against multiple levels based on symphony config (only level0 currently)
+  - Handling of destorying a launchpad
```
Comment (sebastus) [inline readme.md:19]: probably should search the entire code base for the word "destory"

**Review by sebastus — CHANGES_REQUESTED** [review]
Looks very good. A few little things...

File: pkg/landingzone/landingzone.go (hunk 1 above)
Comment (benc-uk) [reply pkg/landingzone/landingzone.go:417]: Done

File: pkg/landingzone/landingzone.go (hunk 3 above)
Comment (benc-uk) [reply pkg/landingzone/landingzone.go:394]: Probably a good idea, there's likey a few other instances of this in the codebase

**Review by sebastus — APPROVED** [review]
lgtm

**Pull request merged**